crates/bevy_input/src/lib.rs

```diff
@@ -69,20 +70,27 @@ impl Plugin for InputPlugin {
                 mouse_button_input_system.label(InputSystem),
             )
             // gamepad
-            .add_event::<GamepadEvent>()
-            .add_event::<GamepadEventRaw>()
+            .add_event::<RawGamepadAxisChangedEvent>()
+            .add_event::<RawGamepadButtonChangedEvent>()
+            .add_event::<GamepadConnectionEvent>()
+            .add_event::<GamepadButtonChangedEvent>()
+            .add_event::<GamepadAxisChangedEvent>()
             .init_resource::<GamepadSettings>()
             .init_resource::<Gamepads>()
             .init_resource::<Input<GamepadButton>>()
             .init_resource::<Axis<GamepadAxis>>()
             .init_resource::<Axis<GamepadButton>>()
             .add_system_to_stage(
                 CoreStage::PreUpdate,
-                gamepad_event_system.label(InputSystem),
+                gamepad_raw_button_event_system.label(InputSystem),
             )
             .add_system_to_stage(
                 CoreStage::PreUpdate,
-                gamepad_connection_system.after(InputSystem),
+                gamepad_raw_axis_event_system.label(InputSystem),
+            )
+            .add_system_to_stage(
+                CoreStage::PreUpdate,
+                gamepad_connection_system.label(InputSystem),
             )
```

Comment:
Quick Question: Is there a reason we don't just use something like [`.add_system_set_to_stage`](https://docs.rs/bevy/latest/bevy/app/struct.App.html#method.add_system_set_to_stage)? Is it to logically break these up into button/axis/connection?

Reply:
I think we should use that pattern here.